cogs5e/charGen.py

```diff
@@ -98,23 +88,26 @@ def chk(m):
             race_response = await self.bot.wait_for('message', timeout=90, check=chk)
         except asyncio.TimeoutError:
             raise InvalidArgument("Timed out waiting for race.")
-        race = await search_and_select(ctx, compendium.fancyraces, race_response.content, lambda e: e.name)
+        race_choices = await get_race_choices(ctx)
+        race = await search_and_select(ctx, race_choices, race_response.content, lambda e: e.name)
 
         await ctx.send(author.mention + " What class?")
         try:
             class_response = await self.bot.wait_for('message', timeout=90, check=chk)
         except asyncio.TimeoutError:
             raise InvalidArgument("Timed out waiting for class.")
-        _class = await search_and_select(ctx, compendium.classes, class_response.content, lambda e: e['name'])
+        class_choices = await available(ctx, compendium.classes, 'class')
```

Comment:
Most of the changes in this file look like this - it presents a list of entities the author (determined by `ctx`) is licensed to view and asks them to select one.

The other changes in this file are mainly pulling the right data attributes after the game data refactor.